### DISCUSSION TITLE
Operator issue with sqlalchemy.func

### DISCUSSION THREAD

**Discussion by Apakottur** [microsoft/pyright#7742] [Q&A] [answered]
Consider the following file:
```python
# test.py
from sqlalchemy import func
import datetime as dt

reveal_type(func.current_date() + dt.timedelta(days=1))
```

Running `mypy test.py` gives:
```text
test.py:5: note: Revealed type is "sqlalchemy.sql.elements.ColumnElement[datetime.date]"
Success: no issues found in 1 source file
```

Running `pyright test.py` gives:
```text
test.py
  test.py:5:13 - error: Operator "+" not supported for types "current_date" and "timedelta" (reportOperatorIssue)
  test.py:5:13 - information: Type of "func.current_date() + dt.timedelta(days=1)" is "Unknown"
1 error, 0 warnings, 1 information
```

The code works as expected, and I would expect pyright to not raise on error on it.
Is it a bug in pyright or an incorrect typing with sqlalchemy?

Versions:
python 3.12
sqlalchemy 2.0.29
pyright 1.1.359
mypy 1.9.0

**Comment by erictraut** [comment]
At first inspection, it looks like this could be a pyright bug, but I'll need to do more analysis to confirm.

I'll create a bug report out of this discussion thread to track it.

**Reply by erictraut** [reply, marked as answer]
Here's a link to the issue: https://github.com/microsoft/pyright/issues/7743
